integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -535,3 +535,35 @@ def test_div_by_zero_ansi(expr):
 @pytest.mark.parametrize('expr', ['1/0', 'a/0', 'a/b'])
 def test_div_by_zero_nonansi(expr):
     _test_div_by_zero(ansi_mode='nonAnsi', expr=expr)
+
+
+def _test_div_overflow(ansi_mode, expr):
+    ansi_conf = {'spark.sql.ansi.enabled': ansi_mode == 'ansi'}
+    data_gen = lambda spark: two_col_df(
+        spark,
+        LongGen(max_val=LONG_MIN),
+        IntegerGen(min_val=-1, max_val=-1), length=1)
+    div_overflow_func = lambda spark: data_gen(spark).selectExpr(expr)
+
+    if ansi_mode == 'ansi':
+        assert_gpu_and_cpu_error(
+            df_fun=lambda spark: div_overflow_func(spark).collect(),
+            conf=ansi_conf,
+            error_message='java.lang.ArithmeticException: Overflow in integral divide')
+    else:
+        assert_gpu_and_cpu_are_equal_collect(div_overflow_func, ansi_conf)
+
+div_overflow_exprs = [
+    'CAST(-9223372036854775808L as LONG) DIV -1',
+    'a DIV CAST(-1 AS INT)',
+    'a DIV b']
+
+@pytest.mark.parametrize('expr', div_overflow_exprs)
+@pytest.mark.xfail(condition=is_before_spark_320(), reason='https://github.com/apache/spark/pull/32260')
```

Comment:
Can we instead have two tests? One that is for 320 and above which verifies that ansi mode is working and another for before 320 that verifies it is not.  If an xfail passes we don't fail the test run.

Reply:
Thanks for review. Updated the tests.